### PR TITLE
Bug FIxes

### PR DESCRIPTION
**Pull request by andredsim**
Covers severel niche crashes Bambu can encounter when certain read types are missing

Updates the warning system so warnings are output as they arise during verbose mode

### CODE REVIEW

File: R/bambu-processReads.R [hunk 1]
```diff
@@ -68,20 +68,25 @@ bambu.processReadsByFile <- function(bam.file, genomeSequence, annotations,
     readGrgList <- prepareDataFromBam(bam.file[[1]], verbose = verbose, use.names = trackReads)
     warnings = c()
```
Comment (cying111) [inline R/bambu-processReads.R:69]: this is not used?

File: R/bambu-processReads_scoreReadClasses.R [hunk 2]
```diff
@@ -34,24 +40,30 @@ scoreReadClasses = function(se, genomeSequence, annotations, defaultModels,
...
 
     model = NULL
-    if (fit){ 
+    rowData(se)$txScore = rowData(se)$txScore.noFit
+    if (fit & length(thresholdIndex)>0){ 
         model = trainBambu(se, verbose = verbose)
         if(returnModel) metadata(se)$model = model
         txScore = getTranscriptScore(rowData(se)[thresholdIndex,], model,
                                  defaultModels)
+                                 print(5)
```
[the remark this answers is not in the document]
Comment (andredsim) [reply R/bambu-processReads_scoreReadClasses.R:58]: nope, thanks for flagging it

File: R/bambu.R [hunk 3]
```diff
@@ -153,7 +152,9 @@ bambu <- function(reads, annotations = NULL, genome = NULL, NDR = NULL,
 
     rm.readClassSe <- FALSE
     readClassList = reads
-    isBamFiles = ifelse(!is(reads, "BamFileList"), all(grepl(".bam$", reads)), FALSE)
+    isRDSs = all(sapply(reads, class)=="RangedSummarizedExperiment")
+    isBamFiles = !isRDSs
+    if(!isRDSs) isBamFiles = ifelse(!is(reads, "BamFileList"), all(grepl(".bam$", reads)), FALSE)
```
Comment (cying111) [inline R/bambu.R:157]: would the following be enough to replace 156-157?
```suggestion
     isBamFiles <- all(grepl(".bam$", reads))
```

**Review by cying111 — APPROVED** [review]

File: R/bambu.R (hunk 3 above)
Comment (andredsim) [reply R/bambu.R:157]: This part of the code is very ugly, unfortunately the problem is using all() on a rangedsummarizedexperiment is very very slow which is why I check first that it isn't the loaded in .rds files.

Comment (jonathangoeke) [reply R/bambu.R:157]: is it possible to provide readclasses directly? I thought it only works with bam files or a list of read class files? (not with read class objects in memory)

Comment (jonathangoeke) [reply R/bambu.R:157]: (I guess the answer is that it is possible...looking through the next code, which is probably helpful to have anyway)

**Review by jonathangoeke — APPROVED** [review]

**Pull request merged**